lib/src/signed_video_h26x_nalu_list.c

```diff
@@ -520,17 +520,23 @@ h26x_nalu_list_get_validation_str(const h26x_nalu_list_t *list)
 }
 
 /* Cleans up the list by removing the validated items. */
-void
+unsigned int
 h26x_nalu_list_clean_up(h26x_nalu_list_t *list)
 {
-  if (!list) return;
+  if (!list) return 0;
 
   // Remove validated items.
+  unsigned int removed_items = 0;
   h26x_nalu_list_item_t *item = list->first_item;
   while (item && item->validation_status != 'P' && !item->need_second_verification) {
+    if (item->validation_status != 'M') {
```

Comment:
What about the statuses "U", "E" and "_", are these all treated as "validated", but might have failed validation?

Reply:
Yes. A NALU has been validated when we have made a decision. That decision could be OK, Not OK, Unknown, Error, or Ignored.
Think of this like a user that blindly adds NALUs one-by-one and wants to know how many of these can I now pass forward for decoding/rendering (if they are authentic).